src/qt/peertablemodel.cpp

```diff
@@ -172,19 +131,52 @@ Qt::ItemFlags PeerTableModel::flags(const QModelIndex &index) const
     return retval;
 }
 
-QModelIndex PeerTableModel::index(int row, int column, const QModelIndex &parent) const
+QModelIndex PeerTableModel::index(int row, int column, const QModelIndex& parent) const
 {
     Q_UNUSED(parent);
-    CNodeCombinedStats *data = priv->index(row);
 
-    if (data)
-        return createIndex(row, column, data);
+    if (0 <= row && row < rowCount() && 0 <= column && column < columnCount()) {
+        return createIndex(row, column, const_cast<CNodeCombinedStats*>(&m_peers_data[row]));
+    }
+
     return QModelIndex();
 }
 
 void PeerTableModel::refresh()
 {
-    Q_EMIT layoutAboutToBeChanged();
-    priv->refreshPeers(m_node);
-    Q_EMIT layoutChanged();
+    interfaces::Node::NodesStats nodes_stats;
+    m_node.getNodesStats(nodes_stats);
+    decltype(m_peers_data) new_peers_data;
+    new_peers_data.reserve(nodes_stats.size());
+    for (const auto& node_stats : nodes_stats) {
+        const CNodeCombinedStats stats{std::get<0>(node_stats), std::get<2>(node_stats), std::get<1>(node_stats)};
+        new_peers_data.append(stats);
+    }
+
+    // Handle a new peer addition or an old peer in a right way. See:
```

Comment:
nit, or something similar
```suggestion
    // Handle new peer addition or removal as suggested in Qt Docs. See:
```

Reply:
Thanks! [Updated](https://github.com/bitcoin-core/gui/pull/164#issuecomment-855996526).